ibis/expr/types/relations.py

```diff
@@ -1040,6 +1069,9 @@ def aggregate(
         metrics: Sequence[ir.Scalar] | None = (),
         by: Sequence[ir.Value] | None = (),
         having: Sequence[ir.BooleanValue] | None = (),
+        grouping_sets=(),
+        rollups=(),
+        cubes=(),
```

Comment:
I don't like exposing these arguments in the `aggregate`. Here's why:

1. I have to bind again, similar to what I am doing in the `group_by` method.
1. Putting the grouping set arguments alongside `by` means there's potential for confusion and mixing: e.g., someone can passing `ibis.cube` into `by`, and/or they can pass it into `cube`. If we ultimately go this route then we'll have to handle that.

I suppose neither of these things is the actual worst, but I'm open to suggestions on how to avoid exposing these additional arguments.

Reply:
It doesn't get away from having to bind again (and actually adds a bit more work), but what if we just have all grouping expressions passed in to `by`?

Reply:
I'll poke at it a bit. It might be easier if there were bound and unbound versions of the rollup/cube/grouping sets objects. Bound would be used internally only, while the user facing APIs are all unbound. The current ones are effectively Unbound.